enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Baremetal-Keepalived-Unicast
+authors:
+  - "@bcrochet"
+  - "@yboaron"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-02-21
+last-updated: 2020-02-21
+status: implementable
+---
+
+# baremetal-keepalived-unicast
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Non cloud platforms use VIPs (Virtual IP) and [Keepalived](https://www.keepalived.org/) to support high availability for API and default Ingress.
+For more information in the matter of VIPs please use the following [link](https://github.com/openshift/enhancements/blob/master/enhancements/network/baremetal-networking.md).
+
+In current implementation, Keepalived is configured to run [VRRP management protocol](https://tools.ietf.org/html/rfc3768#section-3) of the VIP addresses in a multicast mode, but since not all environments have multicast available, this proposal seeks to remedy that.
+
+
+## Motivation
+
+As mentioned above, currently Keepalived sends the VRRP management protocol via multicast. This requires a
+minimal amount of configuration, as the multicast peers do not need to be specified.
+However, there are likely to be environments where multicast has been disabled because
+of network policy. 
+
+Additionally, moving to unicast will also solve the possible collisions when two or more clusters are on the same broadcast/multicast domain. This use case described in more detail in `user story #2` below.
+
+In environments where there are lots of existing virtual routers managing Virtual IPs with multicast messaging or disallow the multicast VRRP traffic completely, it is necessary to configure Keepalived unicast communication  rather than making it an optional configuration. Making it default ensures a successful deployment, so It makes sense to make it the default and if we want to continue to support multicast VRRP, have it as an option.
+
+### Goals
+
+* Allow deployment in networks where multicast VRRP traffic is forbidden or non-functional.
+* No loss of functionality from the current multicast implementation.
+* Prevent VRRP ID collisions during deployment.
```

Comment:
if this is supposed to be changes for all clusters, we need to include a goal that can move all existing clusters to this new model without downtime...

Reply:
Yep, I'll add that goal.
But I'm not sure we can promise zero downtime, even for planned failover we still have the time it takes for the new leader to publish himself.

Reply:
Done